libs/clients/vmst/src/lib/utils.ts

```diff
@@ -8,39 +56,82 @@ export const createWrappedFetchWithLogging = (
   return new Promise((resolve, reject) => {
     fetch(input, init)
       .then(async (response) => {
-        const body = init?.body ? JSON.parse(init?.body as string) : {}
+        let requestBody = init?.body ? JSON.parse(init?.body as string) : {}
 
-        const newInit = {
-          ...init,
-          body: {
-            ...body,
-            applicant: 'hidden',
-            otherParentId: '' ?? 'hidden',
-            email: 'hidden',
-            phoneNumber: 'hidden',
-            paymentInfo: {
-              ...body?.paymentInfo,
-              bankAccount: 'hidden',
-            },
-            attachments: body?.attachments?.length,
+        // Filter known sensitive data
+        // First pick what we need
+        requestBody = pick(requestBody, [
+          'adoptionDate',
+          // 'applicant',
+          'applicationComment',
+          'applicationFundId',
+          'applicationId',
+          'attachments',
+          'dateOfBirth',
+          // 'email',
+          'employers',
+          'expectedDateOfBirth',
+          'otherParentBlocked',
+          // 'otherParentId',
+          'paymentInfo',
+          'periods',
+          // 'phoneNumber',
+          'rightsCode',
+          'status',
+          // 'testData',
+        ])
+        // Then omit the sensitive sub-attributes
+        // requestBody = omit(requestBody, [
+        //   'employers.approverNationalRegistryId',
+        //   'employers.email',
+        //   'paymentInfo.bankAccount',
+        // ])
+        requestBody.employers = requestBody.employers.map(
```

Comment:
Fix error: `Cannot read properties of undefined (reading 'map')`
```suggestion
        requestBody.employers = requestBody.employers?.map(
```